### PR TITLE
enhancement: Add Error Handling for Missing Input Fields

### LINKED ISSUE #7: [Enhancement] Add Error Handling for Missing Input Fields
### Description:
Currently, the extension silently fails when input fields are not found. Add a user notification or console message for better debugging.

### Files Involved:
`background.js`, `content.js`

### Expected Outcome:
Display a message in the console or UI when input fields are not found.

### PR DESCRIPTION
**Pull request by om-3004**
# Fixes #7 (Add Error Handling for Missing Input Fields)

Added an alert that displays a message when the input field is left empty.

### Nothing typed - ChatGPT
Screenshot 2024-10-09 123935
### White Spaces - ChatGPT
Screenshot 2024-10-09 123957
### Multiple empty lines - ChatGPT
Screenshot 2024-10-09 124013
### Nothing types - Gemini
Screenshot 2024-10-09 124037
### White Spaces - Gemini
Screenshot 2024-10-09 124050

### CODE REVIEW

**Comment by om-3004** [comment]
Hello @atj393, Please find and review my PR and let me know if there are any issues.

**Comment by atj393** [comment]
@om-3004 Great work! Thanks for the attached screenshot. I will review all the changes together at the end of the day.
- Question, have you merged the latest changes from the main branch to the forked branch?
-Discussion, can we use a small notification popup instead of an alert? Is it possible? How much effort do you think it will take?

**Comment by om-3004** [comment]
Thank you @atj393 !!

- Yes, I have merged the latest changes from the main branch. And I did checkout a new branch and commit the code changes in that branch. 

Screenshot 2024-10-09 140515

- For the notification popup, we can use WebNotifications API which will not take up much effort. It looks like this:

Screenshot 2024-10-09 153233

- We can also make our custom notification pop-up box with using CSS or we can use any library by installing it from npm.

**Comment by om-3004** [comment]
Hello @atj393 , I've committed the change of showing message by WebNotification API which is auto dismissed after 5 seconds. Please check and let me know if there is something to update/add. Thanks

**Comment by om-3004** [comment]
Here is the screen recording of it:
https://github.com/user-attachments/assets/41315b48-c7e9-4a4e-a554-b96a954ebdb8

**Comment by atj393** [comment]
@om-3004 Great work! Thanks.

I will review the code and merge the request by EOD. Let you know soon!

**Comment by atj393** [comment]
@om-3004 This isn't working for me. I checked out your branch and tested it locally, but the notifications aren't appearing.
Please see the attached video.
[screen-capture.webm](https://github.com/user-attachments/assets/4ecba0ad-a39f-4744-9a5a-2791d6a06ebe)

**Comment by om-3004** [comment]
Hello @atj393 , sorry I think I forgot to add permission for notifications. I have added that manifest file in the latest commit. Please check if that works now.

**Comment by om-3004** [comment]
@atj393 Also the notifications might have been blocked in your browser, in which case the notification won't pop. So you can try by resetting the notification permission to 'default'.

There are 3 permissions for the notifications:
1.  Granted - The notification will show up
2.  Default - We can request for permission (Allow/Block)
3.  Denied - The notification won't pop

So I was thinking in case the permission is denied, maybe we can show alert to the user to change the permission for notifications.

**Comment by atj393** [comment]
@om-3004 Thank you for your response.
I have tried enabling the permissions and also verified that the permission type was 'default' while checking.

I will review it again within 2 days and finalize it. Thanks.

**Comment by atj393** [comment]
@om-3004 I could not able to replicate it again.

**Comment by om-3004** [comment]
@atj393 let me see if I can find something about the notifications.

**Comment by om-3004** [comment]
- @atj393 I've added alert if the notification permission is set to `block`.
- If it is set to `ask/default`, request for permission.
- And it is set to `allow` the notification is poped up.

https://github.com/user-attachments/assets/49aa13b4-a532-429a-81a9-ba56cea98dc1

**Comment by atj393** [comment]
@om-3004 I will check it tonight. Thanks.

**Comment by atj393** [comment]
@om-3004 meantime can you resolve the conflicts?

**Comment by om-3004** [comment]
@atj393 I've merged the main with my feature branch and resolved the merge conflicts. Please check and let me know if there is any issue.

File: background.js [hunk 1]
```diff
@@ -97,31 +97,40 @@ function handleDoubleClick(tab) {
  * This function is injected into the active tab and interacts directly with the DOM of the page.
  */
 function toggleInputStorage() {
-  const inputFieldChatGPT =
```
Comment (atj393) [inline background.js:100]: @om-3004 Why is this selector removed? it is the fix for the bug #4

**Review by atj393 — CHANGES_REQUESTED** [review]

**Comment by om-3004** [comment]
@atj393 I've updated that lines of code, please verify.

**Review by atj393 — APPROVED** [review]
@om-3004 Great Work. I am merging the changes. Thanks for your efforts!

**Pull request merged**